src/screens/Home/ActionButtons.js

```diff
@@ -0,0 +1,380 @@
+import type { NavigationScreenProp } from 'react-navigation';
+
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  accounts: Accounts,
+  baseFiatCurrency: ?string,
+  rates: Rates,
+  balances: BalancesStore,
+  smartWalletFeatureEnabled: boolean,
+  bitcoinFeatureEnabled: boolean,
+  bitcoinBalances: BitcoinBalance,
+  bitcoinAddresses: BitcoinAddress[],
+  switchAccount: (accountId: string, privateKey?: string) => void,
+  resetIncorrectPassword: () => void,
+};
+
+type State = {
+  visibleActionModal: string,
+  receiveAddress: string,
+  showPinModal: boolean,
+  onPinValidAction: ?(_: string, wallet: EthereumWallet) => Promise<void>,
+  isChangingAcc: boolean,
+};
+
+
+const ActionButtonsWrapper = styled.View`
+  padding: 45px 26px 36px;
+  flex-direction: row;
+  justify-content: space-between;
+`;
+
+
+const getModalActionsInfo = (actionType: string) => {
+  switch (actionType) {
+    case ACCOUNT_TYPES.SMART_WALLET:
+      return {
+        title: 'Smart Wallet',
+        paragraph: 'You are able to recover your wallet using another device, i.e. desktop computer.',
+        children: (<LabelBadge label="Recommended" positive containerStyle={{ marginTop: 11 }} />),
+      };
+
+    case ACCOUNT_TYPES.KEY_BASED:
+      return {
+        title: 'Key Wallet',
+        paragraph: 'Needs to be backed up in order to enable Smart Wallet recovery.',
+      };
+
+    case BLOCKCHAIN_NETWORK_TYPES.BITCOIN:
+      return {
+        title: 'Bitcoin Wallet',
+      };
+    default:
+      return {};
+  }
+};
+
+
+class ActionButtons extends React.Component<Props, State> {
+  state = {
+    visibleActionModal: '',
+    receiveAddress: '',
+    showPinModal: false,
+    onPinValidAction: null,
+    isChangingAcc: false,
+  };
+
+  openActionModal = (actionModalType: string) => {
+    this.setState({ visibleActionModal: actionModalType });
+  };
+
+  closeActionModal = (callback: () => void) => {
+    this.setState({ visibleActionModal: '' }, () => {
+      if (callback) callback();
+    });
+  };
+
+  closeReceiveModal = () => {
+    this.setState({ receiveAddress: '' });
+  };
+
+  handleAuthModalClose = () => {
+    const { resetIncorrectPassword } = this.props;
+    resetIncorrectPassword();
+    this.setState({ showPinModal: false });
+  };
+
+  getModalActions = () => {
+    const { visibleActionModal } = this.state;
+    const {
+      rates,
+      accounts: _accounts,
+      balances,
+      baseFiatCurrency,
+      smartWalletFeatureEnabled,
+      bitcoinFeatureEnabled,
+      bitcoinBalances,
+      bitcoinAddresses,
+    } = this.props;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+
+    const keyWallet = _accounts.find(({ type }) => type === ACCOUNT_TYPES.KEY_BASED) || {};
+    const accountsToShow = [keyWallet];
+
+    if (smartWalletFeatureEnabled) {
+      const smartWallet = _accounts.find(({ type }) => type === ACCOUNT_TYPES.SMART_WALLET) || {};
+      accountsToShow.unshift(smartWallet);
```

Comment:
If smartwallet doesn't exist, `{}` gets added as an account, breaking `calculateBalanceInFiat(rates, balances[id], fiatCurrency)` below.

```suggestion
      const smartWallet = _accounts.find(({ type }) => type === ACCOUNT_TYPES.SMART_WALLET);
      if (smartWallet) {
        accountsToShow.unshift(smartWallet);
      }
```